A11y Bug Fix: Screenreader Announcement on `Clear All Data`

ABradham opened the pull request:
Fixed an accessibility bug where screen readers wouldn't report that all users had been cleared from the table after the `Clear All Data` button on the `Authentication` screen was pressed.

Fixed by adding a [visually hidden](https://www.a11yproject.com/posts/how-to-hide-content/) [aria alert](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role) that notifies screen readers without changing the UI.

This change also required the addition of a new field to the `AuthState` redux store.

christhompsongoogle commented on src/components/Auth/UsersCard/table/UsersTable.tsx:
```diff
@@ -169,6 +172,7 @@ export const UsersTable: React.FC<React.PropsWithChildren<UsersTableProps>> = ({
           </DataTableBody>
         </DataTableContent>
       </DataTable>
+      {shouldShowNukedUsers && <UsersCleared />}
```
nit: let's remove the nuked terminology for "cleared" or "clearedAll" for clarity

ABradham replied:
For sure! I also found the term "nuked" kind of gauche, but kept it to follow with the action definitions in `src/store/auth/actions.tsx`. I'll change the names to `cleared`!

ABradham commented:
Added `alertText` field to redux store and refactored `UsersCleared` component to fetch props directly from store (no passing down props from parent component or conditional rendering from parent component). This pattern seems extensible to other alerts; It might make sense to rename the component from  `UsersCleared` to `UserAlerts` so that other actions that should notify screen readers (creating users, fetching data from the store, starting the emulator, etc.) can take advantage of this component / pattern as well.

christhompsongoogle commented on src/components/Auth/UsersCard/table/UsersCleared.test.tsx:
```diff
@@ -0,0 +1,27 @@
+ */
+
+import { render } from '@testing-library/react';
+import React from 'react';
```
This bit shouldn't be necessary anymore, after react 17 or so

ABradham replied:
I get this error on the `<UsersCleared />` component when I remove the import: `'React' refers to a UMD global, but the current file is a module. Consider adding an import instead.`
 
I'm not entirely sure why importing React is necessary here, but it seems to break the build when I remove it.

christhompsongoogle commented on src/components/Auth/UsersCard/table/UsersCleared.tsx:
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2021 Google LLC
```
nit: 2023

christhompsongoogle replied:
Please update the other headers x3

christhompsongoogle commented on src/components/Auth/UsersCard/table/UsersCleared.tsx:
```diff
@@ -0,0 +1,41 @@
+
+export const UsersCleared: React.FC<
+  React.PropsWithChildren<UsersClearedProps>
+> = ({ alertText }) => {
```
format

ABradham replied:
This seems to be the way that the `Prettier` formatter renders this unwieldy component definition; This is the output after running the `npm run format` rule.

christhompsongoogle commented on src/store/auth/reducer.tsx:
```diff
@@ -35,6 +36,7 @@ export const authReducer = createReducer<AuthState, Action>(INIT_STATE)
         ...users,
         payload.user,
       ]);
+      draft.alertText = ''; // Maybe have an alert for "User Created"?
```
Best to use TODOs:

// TODO(abradham): Consider adding an alert for "User Created"
